### ISSUE TITLE
Dependency issue

### ISSUE THREAD

**Issue by hamid-eis** [nestjs/azure-storage#313]
## I'm submitting a...

<pre><code>
[ ] Regression 
[*] Bug report
[ ] Feature request
[ ] Documentation issue or request
[ ] Support request => Please do not submit support request here, instead post your question on Stack Overflow.
</code></pre>

## Current behavior

When installing I get this error
<pre><code>
npm ERR! Found: @nestjs/common@9.1.5
npm ERR! node_modules/@nestjs/common
npm ERR!   @nestjs/common@"^9.1.4" from the root project
npm ERR! 
npm ERR! Could not resolve dependency:
npm ERR! peer @nestjs/common@"^8.0.0" from @nestjs/azure-storage@3.0.0
npm ERR! node_modules/@nestjs/azure-storage
npm ERR!   @nestjs/azure-storage@"^3.0.0" from the root project
npm ERR! 
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force, or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
</code></pre>

## Expected behavior

It should install without any error. I think a pull request is already created to update this package's dependencies to nestjs latest version.

## Environment

<pre><code>
Nest version: 9.0.0

 
For Tooling issues:
- Node version: 16.18.0  
- Platform: Windows 

</code></pre>

**Comment by CryptoIO** [comment]
Same issue for this

**Comment by sampaiorafael** [comment]
Same here. Any way to bypass or solve this without break everything?

**Comment by hamid-eis** [comment]
I used the --force flag to install dependencies. That resolved it for me.

**Comment by manekinekko** [comment]
Is this related to #329 cc @NikolaPeevski ?

**Comment by NikolaPeevski** [comment]
@manekinekko @kamilmysliwiec Not exactly, #329 is about making the package build, not to make it compatible with the latest nest versions (10 and 11), more specifically
https://github.com/nestjs/azure-storage/blob/master/package.json#L20 these 3.
I'll open a ticket in a bit and put another pull request in a few hours. Tested locally the migration to nest peer 10 and 11 and there doesn't appear to be neither build time breaking changes nor run time (validated by the test cases)

**Comment by manekinekko** [comment]
Thank you for taking the time to explore this.
